diff --git a/java/src/main/java/org/eu/smileyik/luajava/type/LuaTable.java b/java/src/main/java/org/eu/smileyik/luajava/type/LuaTable.java
index 7619796..a9eae83 100644
--- a/java/src/main/java/org/eu/smileyik/luajava/type/LuaTable.java
+++ b/java/src/main/java/org/eu/smileyik/luajava/type/LuaTable.java
@@ -159,7 +159,7 @@ public class LuaTable extends LuaObject implements ILuaCallable, ILuaFieldGettab
      * @throws Exception throw any exception
      */
     public <K, V> Result<Map<K, V>, ? extends Exception> asDeepMap(Class<K> kClass, Class<V> vClass) {
-        return asDeepMap(new HashSet<>(List.of(this)), kClass, vClass);
+        return asDeepMap(new HashSet<>(Collections.singleton(this)), kClass, vClass);
     }
 
     protected <K, V> Result<Map<K, V>, ? extends Exception> asDeepMap(
