diff --git a/java/src/main/java/org/keplerproject/luajava/LuaJavaAPI.java b/java/src/main/java/org/keplerproject/luajava/LuaJavaAPI.java
index a787b9a..c0f2091 100755
--- a/java/src/main/java/org/keplerproject/luajava/LuaJavaAPI.java
+++ b/java/src/main/java/org/keplerproject/luajava/LuaJavaAPI.java
@@ -136,7 +136,7 @@ public final class LuaJavaAPI {
         Object ret;
         try {
             boolean isStatic = Modifier.isStatic(method.getModifiers());
-            if (!method.canAccess(isStatic ? null : obj)) {
+            if (!method.isAccessible()) {
                 method.setAccessible(true);
             }
 
@@ -226,7 +226,7 @@ public final class LuaJavaAPI {
             throw new LuaException("Error accessing field " + fieldName);
         }
         // checkField method already checked the obj can access this field or not.
-        if (!field.canAccess(Modifier.isStatic(field.getModifiers()) ? null : obj)) {
+        if (!field.isAccessible()) {
             field.setAccessible(true);
         }
 
@@ -361,7 +361,7 @@ public final class LuaJavaAPI {
         if (field == null) return 0;
         try {
             Object ret;
-            if (field.canAccess(Modifier.isStatic(field.getModifiers()) ? null : obj)) {
+            if (field.isAccessible()) {
                 ret = field.get(obj);
             } else if (!luaStateFacade.isIgnoreNotPublic()) {
                 field.setAccessible(true);
